`pylint` crashed with a `IndexError`

### Bug description

I was writing code to process a file and accidentally made an error in it -- I forgot the argument of the `next` function. On checking such a pylint crash code. I tried to create an MRE, it turned out the following:
```python
def data(filename):
    with open(filename) as file:
        next() # skip header

        for line in file:
            yield line
```

if I replace `next()` to `next(file)` then code does not crash. If i remove `for` loop pylynt also does not crash.


Information from `pylint-crash-*.txt` file:
pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 790, in _lint_file
    check_astroid_module(module)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1060, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1110, in _check_astroid_module
    walker.walk(node)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  [Previous line repeated 1 more time]
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1071, in visit_call
    self._check_raising_stopiteration_in_generator_next_call(node)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1148, in _check_raising_stopiteration_in_generator_next_call
    and not _looks_like_infinite_iterator(node.args[0])
IndexError: list index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 755, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 792, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Command used

```shell
pylint code.py
```


### Pylint output

```shell
************* Module code
code.py:1:0: C0114: Missing module docstring (missing-module-docstring)
code.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
code.py:2:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
Exception on node <Call l.3 at 0x7f6220ee7670> in file '/home/ismagilov/code.py'
Traceback (most recent call last):
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1071, in visit_call
    self._check_raising_stopiteration_in_generator_next_call(node)
  File "/home/ismagilov/.local/lib/python3.10/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1148, in _check_raising_stopiteration_in_generator_next_call
    and not _looks_like_infinite_iterator(node.args[0])
IndexError: list index out of range
code.py:1:0: F0002: code.py: Fatal error while checking 'code.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/ismagilov/.cache/pylint/pylint-crash-2022-11-23-12-43-53.txt'. (astroid-error)

-----------------------------------
Your code has been rated at 0.00/10
```


### Expected behavior

pylint does not crash

### Pylint version

```shell
$ pylint --version
pylint 2.15.5
astroid 2.12.12
Python 3.10.8 (main, Oct 13 2022, 21:13:48) [GCC 12.2.0]
```


### OS / Environment

GNU/Linux

--- a/doc/whatsnew/fragments/7828.bugfix
+++ b/doc/whatsnew/fragments/7828.bugfix
@@ -1,0 +1,3 @@
+Fixes a crash in ``stop-iteration-return`` when the ``next`` builtin is called without arguments.
+
+Closes #7828

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -1135,6 +1135,11 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             # A next() method, which is now what we want.
             return
 
+        if len(node.args) == 0:
+            # handle case when builtin.next is called without args.
+            # see https://github.com/PyCQA/pylint/issues/7828
+            return
+
         inferred = utils.safe_infer(node.func)
 
         if (

--- a/tests/functional/s/stop_iteration_inside_generator.py
+++ b/tests/functional/s/stop_iteration_inside_generator.py
@@ -175,3 +175,13 @@ def other_safeiter(it):
     it = iter(it)
     while True:
         yield next(1, 2)
+
+def data(filename):
+    """
+    Ensure pylint doesn't crash if `next` is incorrectly called without args
+    See https://github.com/PyCQA/pylint/issues/7828
+    """
+    with open(filename, encoding="utf8") as file:
+        next() # attempt to skip header but this is incorrect code
+        for line in file:
+            yield line

